src/celeritas/user/detail/StepGatherExecutor.hh

```diff
@@ -132,12 +146,44 @@ StepGatherExecutor<P>::operator()(celeritas::CoreTrackView const& track)
         SGL_SET_IF_SELECTED(points[P].dir, geo.dir());
         SGL_SET_IF_SELECTED(points[P].volume_id,
                             geo.is_outside() ? VolumeId{} : geo.volume_id());
+
+        if (this->params.selection.points[P].volume_instance_ids)
+        {
+            auto dst = [this,
+                        &vid = this->state.data.points[P].volume_instance_ids,
+                        tid = track.track_slot_id()] {
+                // Destination size
+                size_type const size = this->params.volume_instance_depth;
+                size_type offset = tid.unchecked_get() * size;
+                auto all_ids = vid[AllItems<VolumeInstanceId>{}];
+                return all_ids.subspan(offset,
+                                       this->params.volume_instance_depth);
+            }();
+
+            // Fill every level from the geometry
+            size_type depth = geo.level().unchecked_get() + 1;
+            CELER_ASSERT(depth <= dst.size());
+            geo.volume_instance_id(dst.first(depth));
+            if constexpr (CELERITAS_DEBUG)
```

Comment:
Now that we have a separate configure option for device debug assertions, should we be checking `CELERITAS_DEVICE_DEBUG` in addition to `CELERITAS_DEBUG` in case only the former is enabled? Is there any reason we wouldn't want debug to always be on when device debug is? (Wondering about this more generally since there are several other places we do this in host/device code, e.g. [killing stuck tracks](https://github.com/celeritas-project/celeritas/blob/5a0178e78d8a4b6b68db1ab5890653a3bdea7f12/src/celeritas/global/alongstep/detail/PropagationApplier.hh#L113-L142).)

Reply:
Oh that's a really good point... should we require `CELERITAS_DEBUG=ON` to enable `CELERITAS_DEVICE_DEBUG`? Or define `CELER_DEBUG = (CELERITAS_DEBUG && !CELER_DEVICE_COMPILE)  || (CELERITAS_DEVICE_DEBUG && CELER_DEVICE_COMPILE)`? I'll update this use case and perhaps we can address the broader issue in a follow-on?

Reply:
I think requiring `CELERITAS_DEBUG` to enable device debug makes sense... I can't think of a good reason for having only the latter enabled.